plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -145,3 +149,20 @@ func getPriorityFunctionConfigs(keys util.StringSet) ([]algorithm.PriorityConfig
 	}
 	return configs, nil
 }
+
+var validName = regexp.MustCompile("^[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])$")
+
+func validateAlgorithmName(name string) error {
+	valid := validName.MatchString(name)
+	if !valid {
+		return fmt.Errorf("algorithm name %v does not matches the valid name regexp %v.", name, validName)
+	}
+	return nil
+}
+
+func fatalValidateAlgorithmName(name string) {
```

Comment:
optional nit: validateAlgorithmNameOrDie would be more consistent with the rest of the codebase, I think.